src/osgPlugins/dae/daeRSceneObjects.cpp

```diff
@@ -361,7 +361,7 @@ osg::Node* daeReader::processLight( domLight *dlight )
     lightmodel->setAmbientIntensity(osg::Vec4(0.0f,0.0f,0.0f,1.0f));
     _rootStateSet->setAttributeAndModes(lightmodel, osg::StateAttribute::ON);
 
-    osg::LightSource* lightsource = new osg::LightSource();
```

Comment:
I have merged this change but code all paths and all calls to processLight correct handle the memory so the detection as leak is a false positive.

Reply:
Correct me if I'm wrong but if you return before freeing the memory like:
https://github.com/jtorresfabra/osg/blob/71732e55f2311d9614992a2217424e96907b5d3e/src/osgPlugins/dae/daeRSceneObjects.cpp#L381 
or 
https://github.com/jtorresfabra/osg/blob/71732e55f2311d9614992a2217424e96907b5d3e/src/osgPlugins/dae/daeRSceneObjects.cpp#L398

aren't the code leaking?

Reply:
You are right.  I am sure I did look over the code looking for early returns, but clearly missed them.  So in this case cppcheck didn't produce a false positive.

As a general rule it's far better to use the ref_ptr<> ptr and ptr.release() approach that you implemented rather than using C pointers just in case the flow of control changes as the code evolves, or in the presence of exceptions.  Most of the new code I write these days uses ref_ptr<> for these reasons.